File: legds/fun2-1.3/c/main.c
#include "main.h"

int main() {
	int a,n;

	printf("Please enter the first element\n");
	scanf("%d",&a);

	printf("Please enter number of sum\n");
	scanf("%d",&n);

	int sum;
	sum = ArraySum(a,n);
	printf("Result of sum = %d\n",sum);

	int *arr = ArraySum0(a, n);

	if(arr[n] > 0) {
		printf("%d",arr[n]);
	}
	for(int i = n-1;i >= 0;i --) {
		char c = (i == 0)?('\n'):('\0');
		printf("%d%c",arr[i],c);	
	}

	int *arr0 = ArraySum1(a, n);

	if(arr0[n] > 0) {
		printf("%d",arr0[n]);
	}
	for(int i = n-1;i >= 0;i --) {
		char c = (i == 0)?('\n'):('\0');
		printf("%d%c",arr0[i],c);	
	}
	return 0;
}

File: legds/fun2-1.3/java/Main.java
import java.util.Scanner;

import src.ArraySum;

public class Main{

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);

		System.out.printf("Please enter the first element%n");
		int a = scan.nextInt();
		System.out.printf("Please enter number of sum%n");
		int n = scan.nextInt();

		int sum = ArraySum.araySum(a, n);
		System.out.printf("Result of sum = %d%n",sum);
		
		int[] arr = ArraySum.arraySum0(a, n);

		if(arr[arr.length - 1] == 1) {
			System.out.printf("1");
		}
		for(int i = arr.length-2;i >= 0;i --) {
			char c = (i == 0)?('\n'):('\0');
			System.out.printf("%d%c",arr[i],c);
		}
		
		int[] arr0 = ArraySum.arraySum1(a, n);

		if(arr0[arr0.length - 1] == 1) {
			System.out.printf("1");
		}

		for(int i = arr0.length-2;i >= 0;i --) {
			char c = (i == 0)?('\n'):('\0');
			System.out.printf("%d%c",arr0[i],c);
		}

		scan.close();
	}
}
